streaming/java/streaming-api/src/main/java/org/ray/streaming/api/stream/DataStreamSource.java

```diff
@@ -27,14 +32,14 @@ public DataStreamSource(StreamingContext streamingContext, SourceFunction<T> sou
    * @param <T>     The type of source data.
    * @return A DataStreamSource.
    */
-  public static <T> DataStreamSource<T> buildSource(
+  public static <T> DataStreamSource<T> fromCollection(
       StreamingContext context, Collection<T> values) {
     return new DataStreamSource(context, new CollectionSourceFunction(values));
   }
 
   @Override
   public DataStreamSource<T> setParallelism(int parallelism) {
-    this.parallelism = parallelism;
+    super.setParallelism(parallelism);
     return this;
```

Comment:
for return this in subclass there is actually a pattern but let's do it as a follow up. The result will be we don't have to override subclass method just for return  `this` in correct type.

Reply:
I thought about this when I change those methods to call super class method. I intended refactor into `recursive self-bounding generics` like `Stream<STREAM_TYPE extends Stream<STREAM_TYPE, T>, T>`. Since it's comment here, maybe we can do it in this PR

Reply:
Just checked in [this](https://github.com/netty/netty/blob/660759b99757748dc416b52028b31f97d80f7d08/transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java#L98-L101). Thanks, it't pretty useful.